plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreState.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ThriftHiveMetastoreState
```

Comment:
It doesn't describe "metastore state" (i read "state" as something that usually can change over time)
It describes "knowledge about metastore" or "metastore capabilities"

The "capabilities" is especially suitable for the metastoreSupportsDateStatistics field (to be moved to this class)
The "knowledge" fits better for the alternative calls...

Let's call this `ThriftHiveMetastoreIntrospection` 
and let's call fields/variables `metastoreIntrospection`.